crates/core/tedge/src/cli/connect/command.rs

```diff
@@ -113,7 +113,14 @@ impl Command for ConnectCommand {
         if self.offline_mode {
             println!("Offline mode. Skipping connection check.\n");
         } else {
-            match self.check_connection(config) {
+            let mut retries = 0;
+            // AWS IoT Just In Time Provisioning (JITP) uses the first connection
+            // to create the "Thing", so the first connection attempt can fail, but retrying
+            // will give it a higher chance of success
+            if self.cloud == Cloud::Aws {
+                retries = 4;
+            }
+            match self.check_connection_with_retries(config, retries) {
```

Comment:
Rather than adding a specific case here, it would be better to add a `retries` setting to [BridgeConfig](https://github.com/thin-edge/thin-edge.io/blob/4d780acc0862997fc73797982f7c7d4174afd725/crates/core/tedge/src/bridge/config.rs#L12) and to introduce the specific value [there](https://github.com/thin-edge/thin-edge.io/blob/4d780acc0862997fc73797982f7c7d4174afd725/crates/core/tedge/src/bridge/aws.rs#L21)

```suggestion
            match self.check_connection_with_retries(config, bridge_config.retries) {
```

Reply:
Resolved by 44e7d66343e70d7792fcfd4e35e450389db7694c